internal/cfgplugins/interface.go

```diff
@@ -37,15 +40,459 @@ const (
+		switch p.PMD() {
+		case ondatra.PMD400GBASEZR, ondatra.PMD400GBASEZRP:
+			ethIndexes[p.Name()] = 40000 + uint32(idx)
+		case ondatra.PMD800GBASEZR, ondatra.PMD800GBASEZRP:
+			ethIndexes[p.Name()] = 80000 + uint32(idx)
+		default:
+			t.Fatalf("Unsupported PMD type for %v", p.PMD())
+		}
+	}
+	return ethIndexes
+}
+
+// UpdateAllConfig configures all the ports.
+func UpdateAllConfig(t *testing.T, dut *ondatra.DUTDevice, frequency uint64, targetOpticalPower float64, operationalMode uint16) {
+	t.Helper()
+	// Set the OC parameters according to the PMD type and operational mode.
+	var (
+		portSpeed           oc.E_IfEthernet_ETHERNET_SPEED
+		numPhysicalChannels uint8
+		rateClass           oc.E_TransportTypes_TRIBUTARY_RATE_CLASS_TYPE
+		tribProtocol        oc.E_TransportTypes_TRIBUTARY_PROTOCOL_TYPE
+		allocation          float64
+		otnIndexes          = AssignOTNIndexes(t, dut)
+		ethIndexes          = AssignETHIndexes(t, dut)
+	)
+	// Create a batch to configure all the OC paths.
+	batch := &gnmi.SetBatch{}
+	for _, p := range dut.Ports() {
+		switch p.PMD() {
+		case ondatra.PMD400GBASEZR, ondatra.PMD400GBASEZRP:
+			portSpeed = oc.IfEthernet_ETHERNET_SPEED_SPEED_400GB
+			numPhysicalChannels = 8
+			rateClass = oc.TransportTypes_TRIBUTARY_RATE_CLASS_TYPE_TRIB_RATE_400G
+			tribProtocol = oc.TransportTypes_TRIBUTARY_PROTOCOL_TYPE_PROT_400GE
+			allocation = 400
+		case ondatra.PMD800GBASEZR, ondatra.PMD800GBASEZRP:
+			switch operationalMode {
+			case 1, 2:
+				portSpeed = oc.IfEthernet_ETHERNET_SPEED_SPEED_800GB
+				numPhysicalChannels = 8
+				rateClass = oc.TransportTypes_TRIBUTARY_RATE_CLASS_TYPE_TRIB_RATE_800G
+				tribProtocol = oc.TransportTypes_TRIBUTARY_PROTOCOL_TYPE_PROT_800GE
+				allocation = 800
+			case 3, 4:
+				portSpeed = oc.IfEthernet_ETHERNET_SPEED_SPEED_400GB
+				numPhysicalChannels = 4
+				rateClass = oc.TransportTypes_TRIBUTARY_RATE_CLASS_TYPE_TRIB_RATE_400G
+				tribProtocol = oc.TransportTypes_TRIBUTARY_PROTOCOL_TYPE_PROT_400GE
+				allocation = 400
+			default:
+				t.Fatalf("Unsupported operational mode for %v: %v", p.PMD(), operationalMode)
+			}
+		default:
+			t.Fatalf("Unsupported PMD type for %v", p.PMD())
+		}
+		// Get the hardware port, transceiver and optical channel names.
+		hwPort := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).HardwarePort().State())
+		transceiverName := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).Transceiver().State())
+		och := components.OpticalChannelComponentFromPort(t, dut, p)
+		updateInterfaceConfig(batch, p, portSpeed)
+		updateHWPortConfig(batch, p, hwPort, portSpeed, numPhysicalChannels)
+		updateOpticalChannelConfig(batch, och, frequency, targetOpticalPower, operationalMode)
+		updateOTNChannelConfig(batch, dut, och, otnIndexes[p.Name()], ethIndexes[p.Name()], allocation, tribProtocol)
+		updateETHChannelConfig(batch, dut, p, transceiverName, otnIndexes[p.Name()], ethIndexes[p.Name()], allocation, rateClass, tribProtocol)
+	}
+	batch.Set(t, dut)
```

Comment:
The batch.Set should be in the _test.go file.  The cfgplugin just just generate/update the Batch query.